### DISCUSSION TITLE
Should Livewire.find() proxy the mount method?

### DISCUSSION THREAD

**Discussion by timrspratt** [livewire/livewire#3119] [7. Bugs (archived)]
More of a typing out loud here, but should `mount` be able to be called from JS? Does this allow for re-hydration of the component with different "initial" properties that could cause issues / be used maliciously?

```
Livewire.find('z2fDP4p2hMDNezwycajl').mount('123', '123')
```

**Comment by joshhanley** [comment]
@timrspratt my first though was, no that shouldn't be possible.

But did you test that and it did work? If so, you probably should have said that in your message 😆

 I just tested it and you can call mount and pass different properties to it, which I didn't expect, and I don't think should be possible in my opinion!!

**Comment by timrspratt** [comment]
Sorry @joshhanley I should have been more thorough, usually I over-explain and send people to sleep 😂

I have tested this, on my own and other production projects (ones that aren't mine), and it does execute. 

I agree that it feels like this shouldn't be allowed, and that mount should only be able to be called from the server code, and not from the JS proxy.

**Comment by joshhanley** [comment]
@timrspratt no worries!

Yep, cool, just wanted to make sure you also found that when testing and that it wasn't just me haha, and that it was unexpected.

**Comment by calebporzio** [comment]
Good call @timrspratt!

Yeah, I agree this shouldn't be allowed. In fact, all lifecycle hooks should be blocked from front-end calls.

Would one of you mind PRing this? Thanks!
